src/coreclr/nativeaot/Runtime/unix/unixasmmacrosamd64.inc

```diff
@@ -318,7 +320,7 @@ C_FUNC(\Name):
 DEFAULT_FRAME_SAVE_FLAGS = PTFF_SAVE_ALL_PRESERVED + PTFF_SAVE_RSP
 
 .macro PUSH_COOP_PINVOKE_FRAME trashReg
-    push_nonvol_reg rbp                         // push RBP frame
+    push_nonvol_reg rbp                         // push RBP frame  // TODO: do we need this? not on windows.
```

Comment:
Regarding the added TODO: I believe this is needed for unwindability. I would think that on Windows, it is needed too.

Reply:
These macros are used to call GC helpers, not the user code, so I assume unwindability here would be useful for crash dumps and such. 
The Windows version seems to be using RSP-based frame style, which makes it simpler as RBP is just a nonvol register. I think the pattern is unwindable, at least as much as needed for debugging.

I will keep the TODO here. I will be doing some cleanup of assembly helpers. I think we have a good deal of redundancy and will take a look at this as a part of cleanup. 
Ideally we should use the same code.

Reply:
I see. Thanks! Saves me time to figure if the alternative works with debuggers (I now know it does not).

I will change Windows to use the same code then (in a separate change). Even if Windows is resilient to this, it is nicer to have the same code where possible.